temp_utility_scripts/remove_incomplete_attachments.py

```diff
@@ -117,19 +124,32 @@ def remove_unattached_image_paths(self, remove_assoc=True):
 
             # this will work well only after the first function which
             # eliminates attachments without collectionobjectattachments
-            for attachment_id in attachment_ids:
-                self.logger.info(f"removing collectionobjectattachment and attachment: {attachment_id[0]}")
-                attachment_id_test_count.append(attachment_id[0])
-                sql = f"""DELETE FROM collectionobjectattachment WHERE AttachmentID = {attachment_id[0]};"""
-                self.sql_csv_tools.insert_table_record(sql=sql)
-                sql = f"""DELETE FROM attachment WHERE AttachmentID = {attachment_id[0]};"""
-                self.sql_csv_tools.insert_table_record(sql=sql)
+            if attachment_ids:
+                for attachment_id in attachment_ids:
+                    self.logger.info(f"removing collectionobjectattachment and attachment: {attachment_id[0]}")
+                    attachment_id_test_count.append(attachment_id[0])
+                    sql = f"""DELETE FROM collectionobjectattachment WHERE AttachmentID = {attachment_id[0]};"""
+                    self.sql_csv_tools.insert_table_record(sql=sql)
+                    sql = f"""DELETE FROM attachment WHERE AttachmentID = {attachment_id[0]};"""
+                    self.sql_csv_tools.insert_table_record(sql=sql)
 
         self.logger.warning(f"Number of barcodes processed: {len(barcode_test_count)}")
         self.logger.warning(f"Number filepaths deleted from image_server: {len(filepath_test_count)}")
         self.logger.warning(f"Number of attachments/collectionobjectattachments deleted: {len(attachment_id_test_count)}")
 
 
```

Comment:
generalist function that can remove a list of internal filepaths from image server , only used if once has an existing list of un-attached filepaths.